### ISSUE TITLE
Problems with DOI import

### ISSUE THREAD

**Issue by vpeil** [LibreCat/LibreCat#564]
For some DOIs (?) the importer shows the correct title (looks like import was successfull). But if you try to edit this record are are forwarded to `/librecat`, because the record is not in the DB! No hints in ~configs~ logs.

Example DOIs:

- 10.1038/s41598-018-30701-5

- 10.1186/s12888-018-1839-z

**Comment by vpeil** [comment]
Strange, today it's working... I leave it open, and will investigate further

**Comment by phochste** [comment]
General problem when no existing DOIs are entered. The site crashes with:

```
// 20190123132920
// http://localhost:5001/librecat/record/import

{
  "error": "Warning caught during route execution: request failed: https://api.crossref.org/works/10.1038/s41598-018-30701-72/agency\n",
  "exception": "Warning caught during route execution: request failed: https://api.crossref.org/works/10.1038/s41598-018-30701-72/agency\n"
}
```

**Comment by vpeil** [comment]
> General problem when no existing DOIs are entered. The site crashes with:
> 
> ```
> // 20190123132920
> // http://localhost:5001/librecat/record/import
> 
> {
>   "error": "Warning caught during route execution: request failed: https://api.crossref.org/works/10.1038/s41598-018-30701-72/agency\n",
>   "exception": "Warning caught during route execution: request failed: https://api.crossref.org/works/10.1038/s41598-018-30701-72/agency\n"
> }
> ```

This shouldn't happen, because the input field has a required attribute.

**Comment by phochste** [comment]
@vpeil I meant when a DOI is entered that doesn't exist at crossref. A patch for this is on its way.

A second problem which is not yet fixed is that the crossref and datacite mapping fixes seem to use a 'blacklist' of fields that can't enter the record. This means that any variantion in the CrossRef record fields get passed the Fix stage and can result in an invalid record (and no record in the database). The `crossref_mapping.fix` and `datacite_mapping.fix` need to be rewritten..and map the fields you know you want to be in the record and delete all the rest.

**Comment by vpeil** [comment]
@phochste PR #684 is ready for merge. Alltogether, this issue should be fixed now.

**Comment by datango** [comment]
Stil problems
10.1055/s-0042-116437

**Comment by phochste** [comment]
@datango there is typo in `fixes/crossref_mapping.fix` on line 95:

```
lookup(r.language, fixes/lookup/cr_lang_iso.csv, default: eng)
```

should be

```
lookup(r.language.0.iso, fixes/lookup/cr_lang_iso.csv, default: eng)
```

I'll create a patch

**Comment by datango** [comment]
Works! Thanks!

**Comment by vpeil** [comment]
fixed by 0d9c353
